core/root/config/config.go

```diff
@@ -112,6 +112,12 @@ func ConfigFromBytesWithEncryptService(data []byte, encryptService controlplane.
 		return nil, err
 	}
 	c.ProvidedEncryptService = encryptService
+
+	// Uses the same encrypt service for node pools
+	for _, p := range c.NodePools {
+		p.ProvidedEncryptService = encryptService
+	}
```

Comment:
This is just to confirm, but this is required to pass one of newly added tests, right?